cms/djangoapps/contentstore/api/views/course_validation.py

```diff
@@ -123,11 +148,20 @@ def _assignments_validation(self, course):
         )
 
         return dict(
-            total_number=len(assignments),
-            total_visible=len(visible_assignments),
-            num_with_dates=num_with_dates,
-            num_with_dates_after_start=num_with_dates_after_start,
-            num_with_dates_before_end=num_with_dates_before_end,
+            all=dict(
```

Comment:
By moving these fields into an `all` block, it technically would be a breaking change to the API. I will let @nasthagiri speak to the proper way of handling this within the edX system both in consideration that I believe we're the only consumer of this API currently as well as how to handle it with regards to APIs with more consumers.